### ISSUE TITLE
bug: Audio playing from other App does not return to full volume after finishing speaking

### ISSUE THREAD

**Issue by Solarus8** [capacitor-community/text-to-speech#138]
**Plugin version:**
 "@capacitor-community/text-to-speech": "^5.0.0"

**Platform(s):**
iPhone 15 Pro Max, iOS 18.0.1

**Current behavior:**
When playing other audio (music app - like Pandora) the volume of that audio does not return to full volume after it is attenuated during the text-2-speech content is finished playing

**Expected behavior:**
The audio of any other app will return to full volume after the text-2-speech content is finished playing.

**Steps to reproduce:**
Open Pandora (or other audio app), open capacitor app with text-2-speech, play text, notice volume is about halved when text is read (this is good) and then the volume of Pandora stays at halved volume until the app is sent to the background or closed.

**Comment by robingenz** [comment]
Would you be willing to create a PR?

**Comment by flexiblefactory** [comment]
I think something like this could be the fix:

```
   // Reset the audio session category to normal after TTS finishes
    let audioSession = AVAudioSession.sharedInstance()
    try? audioSession.setCategory(.ambient, options: [])
```

**Comment by Solarus8** [comment]
Great! Did you wrap that into a capacitor plugin, will you issue a PR?

**Comment by flexiblefactory** [comment]
It is a plugin currently and works as a workaround. I do think it would make sense for the changes above to be integrated into the TTS plugin.

**Comment by physxP** [comment]
> I think something like this could be the fix:
> 
> ```
> // Reset the audio session category to normal after TTS finishes
> 
>     public func speechSynthesizer(_ synthesizer: AVSpeechSynthesizer, didCancel utterance: AVSpeechUtterance) {
>         // Reset session when speech cancelled
>          try AVAudioSession.sharedInstance().setActive(false)
>          self.resolveCurrentCall()
>     }
> 
>     public func speechSynthesizer(_ synthesizer: AVSpeechSynthesizer, didFinish utterance: AVSpeechUtterance) {
>         // Reset session when speech finishes
>         try AVAudioSession.sharedInstance().setActive(false)
>         self.resolveCurrentCall()
>     }
> ```

I believe this causes the UI stuttering if setActive is done during a UI animation or when for example Camera Preview is running. Doing this in the background queue introduces lag of around 300 to 500  ms depending on the device. I believe that something like start or stop method would be more appropriate so it's in the control of the user: https://github.com/capacitor-community/text-to-speech/issues/133

**Comment by flexiblefactory** [comment]
The other aspect to this is that the plugin also sets active `true` at startup time, causing other audio to "duck" as soon as the plugin loads (and then ducking persists until the app quits).

**Comment by physxP** [comment]
> The other aspect to this is that the plugin also sets active `true` at startup time, causing other audio to "duck" as soon as the plugin loads (and then ducking persists until the app quits).

That's true. So, if setActive method is configurable from TS side then think it could fix all of these issues at once?
This could be an API breaking change if not implemented in the most efficient manner

**Comment by hugocxl** [comment]
any updates on this? 🙏

**Comment by robingenz** [comment]
@hugocxl Feel free to submit a PR or check out this [Speech Synthesis](https://capawesome.io/plugins/speech-synthesis/) plugin where the problem does not occur.

**Comment by hugocxl** [comment]
Hi @robingenz!
We have added a few lines to adjust the plugin. In `AppDelete.swift`:

```swift
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        DispatchQueue.main.async {
            try? AVAudioSession.sharedInstance().setCategory(.playback, mode: .default, options: .mixWithOthers)
            try? AVAudioSession.sharedInstance().setActive(false, options: .notifyOthersOnDeactivation)
        }
        return true
    }
```

It modifies the behaviour a bit, not decreasing the volume from other apps at all, but it works for us.
Not sure if that's what you'd expect for the plugin, in that case I can open a PR, sure.

**Comment by robingenz** [comment]
@hugocxl No, unfortunately not. The plugin should handle it differently.

**Comment by laurens** [comment]
I believe the issue can be fixed by completely removing the plugin’s custom `AVAudioSession`  
and setting instead

    self.synthesizer.usesApplicationAudioSession = false

https://developer.apple.com/documentation/avfaudio/avspeechsynthesizer/usesapplicationaudiosession

> If you set this value to [false](https://developer.apple.com/documentation/Swift/false), **the system creates a separate audio session to automatically manage speech, interruptions, and mixing and ducking the speech with other audio sources**.

---

I’ve tested it iOS 18.5 and 14.8 and the behaviour is the same to me:

- other app’s audio is ducked
- a few seconds after end of speech synthesis, it implicitly deactivates, resuming the other app’s volume 


---

Unless someone adds some context on another reason/need for custom audio session, I’d be happy to provide a PR with the change

**Comment by robingenz** [comment]
@laurens Thank you! PRs are always welcome.
